### PR TITLE
feat(roadmap): added roadmap for wasmCloud

### PR DESCRIPTION
**Pull request by brooksmtownsend**
This PR provides motivation and context around https://github.com/orgs/wasmCloud/projects/7/views/3

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
Checks are lying, had to reauthenticate, preview here https://deploy-preview-137--dreamy-golick-5f201e.netlify.app/

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+date: 2023-08-02T00:00:00+00:00
+description: "wasmCloud Development Roadmap"
+sidebar_position: 6
+---
+
+:::info
+This roadmap page is a snapshot of the current state of the project. It is not a guarantee of future direction or features. It is a living document and will change as the project evolves. You can view the most current version of this roadmap on [GitHub Projects](https://github.com/orgs/wasmCloud/projects/7/views/3).
+:::
+
+wasmCloud is a constantly evolving project. The feature roadmap and release schedule aren't rigid at this time, instead adopting a "Now, Next, Later" format to indicate when efforts are available for work. In addition to the higher-level roadmap on [GitHub Projects](https://github.com/orgs/wasmCloud/projects/7/views/3), wasmCloud is under constant development for smaller improvements, bug fixes, and documentation. wasmCloud seeks to evolve with the WebAssembly ecosystem, adopting standards and best practices as they become available. We believe this is the best way, after hearing invaluable feedback from contributors and users, to accomplish our goals as a project.
```
Comment (CazJT) [inline docs/roadmap.md:12]: Add comma after instead, then add 'we're':

aren't rigid at this time, instead, we're adopting xxx

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+date: 2023-08-02T00:00:00+00:00
+description: "wasmCloud Development Roadmap"
+sidebar_position: 6
+---
+
+:::info
+This roadmap page is a snapshot of the current state of the project. It is not a guarantee of future direction or features. It is a living document and will change as the project evolves. You can view the most current version of this roadmap on [GitHub Projects](https://github.com/orgs/wasmCloud/projects/7/views/3).
+:::
+
+wasmCloud is a constantly evolving project. The feature roadmap and release schedule aren't rigid at this time, instead, we're adopting a "Now, Next, Later" format to indicate when efforts are available for work. In addition to the higher-level roadmap on [GitHub Projects](https://github.com/orgs/wasmCloud/projects/7/views/3), wasmCloud is under constant development for smaller improvements, bug fixes, and documentation. wasmCloud seeks to evolve with the WebAssembly ecosystem, adopting standards and best practices as they become available. We believe this is the best way, after hearing invaluable feedback from contributors and users, to accomplish our goals as a project.
```
Comment (CazJT) [inline docs/roadmap.md]: Remove Oxford comma after bug fixes

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Goals
+
+Looking at wasmCloud development in the longer-term, we strive to achieve the following goals:
+
+1. **Leverage as many WebAssembly standards as possible (WASI, Wasm components, wasi-cloud, etc)**
```
Comment (CazJT) [inline docs/roadmap.md]: Period after etc - , etc.)

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+1. Provide a seamless developer experience for building, testing, and deploying WebAssembly components
+1. Be the best way to build vendor-less components for WebAssembly applications
+1. Bring joy to distributed computing
+1. Leverage cloud-native standards where possible (OAM, OCI, CloudEvents, etc.)
+
+wasmCloud began in 2019 with the goal of securely connecting capabilities to applications, leveraging WebAssembly to provide a language and platform agnostic unit of compute. This loose coupling of capabilities and applications is the core of wasmCloud's ability to bring joy to distributed computing, allowing application developers to: maintain their business logic and non-functional requirements separately, build without the intricacies of platform-specific dependencies, and seamlessly distribute applications without changing their code.
```
Comment (CazJT) [inline docs/roadmap.md]: Hyphenate platform-agnostic

add word; is at the core of wasmCloud's ability

Split sentence: 

bring joy to distributed computing. This allows application developers to:

File: docs/roadmap.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+1. **Leverage [WebAssembly interface types (wit)](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) for our interfaces:** we can use and contribute to the wide variety of projects that provide language support for WebAssembly components, extending wasmCloud's language support from Rust and TinyGo to include C, [Javascript](https://github.com/bytecodealliance/jco), [Python](https://github.com/dicej/componentize-py), and more as they become available.
+1. **Implement and use [wasi-cloud](https://github.com/WebAssembly/wasi-cloud-core) instead of proprietary wasmCloud interfaces:** letting users bring their own components to wasmCloud instead of requiring them to be built with wasmCloud's SDK.
+1. **Transition feature focus to Rust:** Recently accepted as [ADR-0013](https://github.com/wasmCloud/wasmCloud/blob/main/adr/0013-transition-feature-focus-to-rust.md), we aim to focus our efforts on the wasmCloud Rust runtime project to rapidly conform to standards and offer a portable, stable, high-performance wasmCloud runtime.
```
Comment (connorsmith256) [inline docs/roadmap.md]: Nit: can we title this one something different? To newcomers I'm afraid it means wasmCloud as a whole is only focusing on Rust, vs our primary runtime being written in Rust

**Review by connorsmith256 — APPROVED** [review]
🚀

**Pull request merged**